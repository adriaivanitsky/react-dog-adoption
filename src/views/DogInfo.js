import React from 'react';
import { useEffect, useState } from 'react';
import { getDogById } from '../services/dogData';
import DogDetail from '../components/DogDetail';

export default function DogInfo(props) {
  const id = props.match.params.id;
  const [loading, setLoading] = useState(true);
  const [dog, setDog] = useState([]);
  useEffect(() => {
    const fetchData = async () => {
      const data = await getDogById(id);
      setDog(data);
      setLoading(false);
    };
    fetchData();
  }, [id]);
  if (loading) return <h1>Loading...</h1>;

  return <DogDetail {...dog} />;
}
