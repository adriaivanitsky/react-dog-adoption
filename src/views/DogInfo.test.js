import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import DogInfo from './DogInfo.js';

test('should render a dog and its stats to the page', async () => {
  const container = render(
    <MemoryRouter>
      <DogInfo match={{ params: { id: 2 } }} />
    </MemoryRouter>
  );
  await screen.findByText('gertie', 'the best girl in the whole world');
  expect(container).toMatchSnapshot();
});
