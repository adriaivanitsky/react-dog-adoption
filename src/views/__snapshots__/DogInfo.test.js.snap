// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a dog and its stats to the page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          home 
        </a>
        <a
          href="/admin"
        >
           admin
        </a>
      </header>
      <h1>
        gertie
      </h1>
      <div>
        <img
          src="https://3.bp.blogspot.com/_vB43sFWYD_E/TU9kGWQZv6I/AAAAAAAADtk/otXtGuOyRx4/s1600/Boston_Terrier_IMG_3283blog.jpg"
        />
        <p>
          the best girl in the whole world 
        </p>
        <p>
          boston terrier
        </p>
      </div>
      <a
        href="/dogs/2/edit"
      >
        <button>
          edit
        </button>
      </a>
      <button>
        delete
      </button>
    </div>
  </body>,
  "container": <div>
    <header>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        home 
      </a>
      <a
        href="/admin"
      >
         admin
      </a>
    </header>
    <h1>
      gertie
    </h1>
    <div>
      <img
        src="https://3.bp.blogspot.com/_vB43sFWYD_E/TU9kGWQZv6I/AAAAAAAADtk/otXtGuOyRx4/s1600/Boston_Terrier_IMG_3283blog.jpg"
      />
      <p>
        the best girl in the whole world 
      </p>
      <p>
        boston terrier
      </p>
    </div>
    <a
      href="/dogs/2/edit"
    >
      <button>
        edit
      </button>
    </a>
    <button>
      delete
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
